src/datasets/arrow_dataset.py

```diff
@@ -159,6 +159,51 @@ def version(self):
         return self._info.version
 
 
+class TensorflowDatasetMixIn:
+    def __init__(self):
+        pass
+
+    def to_tf_dataset(self, tokenizer, cols_to_remove, batch_size, shuffle):
```

Comment:
In terms of general design, I would like the tokenizer to be optional. In most cases this won't be necessary.
And in terms of optional argument, better than a tokenizer, we should just take an `data_collator` function that could just take the samples and form a batch from it. This way we can just use the same data collators as in PyTorch with a few tweaks to make tensorflow tensors at the end.

Reply:
Resolved! We now take an optional collate_fn argument, and pad with zeroes if one isn't provided